test/e2e/v4/generate_test.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v4
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	pluginutil "sigs.k8s.io/kubebuilder/v3/pkg/plugin/util"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/ginkgo"
+
+	//nolint:golint
+	//nolint:revive
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/kubebuilder/v3/test/e2e/utils"
+)
+
+// GenerateV2 implements a go/v2 plugin project defined by a TestContext.
+func GenerateV2(kbc *utils.TestContext) {
```

Comment:
Why we are generating a go/v2 here?

Reply:
So we have to 
**Regarding test/e2e/**
- Remove `test/e2e/v4` directory.
- Add a function `GenerateV4` in `test/e2e/v3/generate_test.go` similar to `GenerateV3` but with plugin : go/v4 and version : 4.
- Add Context similar to https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/v3/plugin_cluster_test.go#L98-L129
but with `Root : "plugin go.kubebuilder.io/v4"` in `test/e2e/v3/plugin_cluster_test.go` and use func `GenerateV4` instead in its logic.

Could you inform I am understanding correct here.